envoy/network/connection_handler.h

```diff
@@ -19,6 +21,26 @@
 namespace Envoy {
 namespace Network {
 
+// The thread local registry.
+class LocalRevConnRegistry {
```

Comment:
I suspect you should be able to do this PR with minimal APIs in core Envoy, but instead in your extension directory

Reply:
@alyssawilk  The main set of changes is to support the creation of the ReverseConnectionManager and ReverseConnectionHandler [thread-locally](https://github.com/nutanix/envoyproxy-envoy/blob/af13cddb363be33c4adf913aba6bda88a32257d0/source/extensions/bootstrap/reverse_connection/reverse_conn_thread_local_registry.cc) as implemented using a thread-local registry. The ReverseConnectionManager and ReverseConnectionHandler need to come up [right after the workers are created](https://github.com/nutanix/envoyproxy-envoy/blob/457e1e985a0ef2c0812f809d34aa3add2d029433/source/common/listener_manager/listener_manager_impl.cc#L371) because they are involved in accepting reverse connections from other initiating envoy instances even if the local envoy is not initiating any reverse connections, as long as the [local envoy has the reverse connections bootstrap extension enabled](https://github.com/nutanix/envoyproxy-envoy/blob/457e1e985a0ef2c0812f809d34aa3add2d029433/source/server/server.cc#L820).

So, currently, we check if the reverse connection bootstrap extension is enabled by checking if the singleton is present, and if so, [right after workers are created, we post to each worker's dispatcher](https://github.com/nutanix/envoyproxy-envoy/blob/457e1e985a0ef2c0812f809d34aa3add2d029433/source/server/server.cc#L820) a functor to create the Thread Local Registries (creating the ReverseConnectionManager and ReverseConnectionHandler). This thread local registry is then parked with the Connection Handler and is accessed later wherever we need access to the  ReverseConnectionManager and ReverseConnectionHandler like [here,to initiate reverse conns](https://github.com/nutanix/envoyproxy-envoy/blob/af13cddb363be33c4adf913aba6bda88a32257d0/source/extensions/reverse_connection/active_reverse_connection_listener.cc#L44), [here in the reverse conn filter, which accepts reverse conns](https://github.com/nutanix/envoyproxy-envoy/blob/af13cddb363be33c4adf913aba6bda88a32257d0/source/extensions/filters/http/reverse_conn/reverse_conn_filter.h#L124) and in multiple other places where only the thread local dispatcher is otherwise available to us. Therefore, I have defined abstract classes for LocalRevConnRegistry, RevConnRegistry, etc so that these two entities can be stored in the form of a ThreadLocalRegistry and accessed later. We briefly touched upon this in [this comment](https://github.com/envoyproxy/envoy/issues/33320#issuecomment-2299569235) and on slack. In summary:

1. We have created two thread local entities which get initialized right after the workers are created. They need to be parked somewhere so that they can be accessed later in code paths where only the thread local dispatcher is available. We have parked them with the thread local connection handler and therefore added the abstract classes under Envoy::Network.
2. If the above is not the best approach, what would you suggest?

Reply:
> This thread local registry is then parked with the Connection Handler and is accessed later wherever we need access to the ReverseConnectionManager and ReverseConnectionHandler like here,to initiate reverse conns, here in the reverse conn filter, which accepts reverse conns and in multiple other places where only the thread local dispatcher is otherwise available to us

For filters we can get the slot from  serverFactoryContext from `Server::Configuration::FactoryContext& context` during the filter chain creation, could you try to see if this can avoid the `setConnectionHandler` api changes?